Week3/homework/step2-3.js

```diff
@@ -29,10 +35,19 @@ console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 // Use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line prefer-const
-  let result = '';
-
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  let result;
+  const repeatArr = [];
+  let i = 0;
+  if (i >= 0) {
+    do {
+      repeatArr.push(str);
+      i++;
+    } while (i < num);
+    result = repeatArr.join('');
+    if (num <= 0) {
+      result = '';
+    }
```

Comment:
If you have correctly implemented the test **_before_** entering the loop you no longer need lines 47-49.

Reply:
I think I could not change my focus to this file. that's why every time I don't see a stupid point. 
I thought to type lines 47 and 48 but I couldn't think >= will work for 0 anytime.
I updated with if i > 0, and I removed the 47-49 lines

Reply:
I can see that you are struggling with this. Making the test `i > 0` doesn't help. You need to test `num`. That is the parameter that is passed to your function and that might have a value <= 0. So the test in your `if` statement should be `num > 0`, not `i > 0`.

Does that now make sense?

Reply:
oh why am I like that..
I am sorry for taking that much time of you. 
thanks for understanding. 
and yes. the action is about 'num' not about ' i '. just loop uses ' i ' for counting.